test/suites/test_datetime_arithmetic.py

```diff
@@ -0,0 +1,317 @@
+                                         [case['res_sub']])
+
+
+    datetime_sub_cases = {
+        'date': {
+            'arg_1': tarantool.Datetime(year=2008, month=2, day=3),
+            'arg_2': tarantool.Datetime(year=2010, month=2, day=1),
+            'res': tarantool.Interval(year=-2, month=0, day=2),
+        },
+        'datetime': {
+            'arg_1': tarantool.Datetime(year=2001, month=2, day=3, hour=1, minute=2, sec=30),
+            'arg_2': tarantool.Datetime(year=2002, month=1, day=31, hour=3, minute=0, sec=20),
+            'res': tarantool.Interval(year=-1, month=1, day=-28, hour=-2, minute=2, sec=10),
+        },
+        'datetime_with_nsec': {
+            'arg_1': tarantool.Datetime(year=2001, month=2, day=3, hour=1, minute=2,
+                                        sec=30, nsec=10000000),
+            'arg_2': tarantool.Datetime(year=2002, month=1, day=31, hour=3, minute=0,
+                                        sec=10, nsec=9876543),
+            'res': tarantool.Interval(year=-1, month=1, day=-28, hour=-2, minute=2,
+                                      sec=20, nsec=123457),
+        },
+        'heterogenous': {
+            'arg_1': tarantool.Datetime(year=2001, month=2, day=3, hour=1, minute=2),
+            'arg_2': tarantool.Datetime(year=2001, month=2, day=3, sec=30,
+                                        nsec=9876543),
+            'res': tarantool.Interval(hour=1, minute=2, sec=-30, nsec=-9876543),
+        },
+    }
+
+    def test_python_datetime_sub(self):
+        for name in self.datetime_sub_cases.keys():
+            with self.subTest(msg=name):
+                case = self.datetime_sub_cases[name]
+
+                self.assertEqual(case['arg_1'] - case['arg_2'], case['res'])
+
+    @skip_or_run_datetime_test
+    def test_tarantool_datetime_sub(self):
+        for name in self.datetime_sub_cases.keys():
+            with self.subTest(msg=name):
+                case = self.datetime_sub_cases[name]
+
+                self.assertSequenceEqual(self.con.call('sub', case['arg_1'], case['arg_2']),
+                                         [case['res']])
+
+
+    datetime_sub_different_timezones_case = {
+        'arg_1': tarantool.Datetime(year=2001, month=2, day=3, tz='UTC'),
+        'arg_2': tarantool.Datetime(year=2001, month=2, day=3, tz='MSK'),
+        'res': tarantool.Interval(day=1, hour=-21),
+    }
+
+    def test_python_datetime_sub_different_timezones(self):
+        case = self.datetime_sub_different_timezones_case
+
+        self.assertEqual(case['arg_1'] - case['arg_2'], case['res'])
+
+    @skip_or_run_datetime_test
+    @unittest.expectedFailure # See https://github.com/tarantool/tarantool/issues/7698
+    def test_tarantool_datetime_sub_different_timezones(self):
+        case = self.datetime_sub_different_timezones_case
+
+        self.assertSequenceEqual(self.con.call('sub', case['arg_1'], case['arg_2']),
+                                         [case['res']])
```

Comment:
Wil the testsl fail on a new versions of Tarantool with fixes? It seems better to skip the test.

Reply:
Yes, it will. I think it's better to wait until the bug would be fixed and then replace xfail with conditional xfail based on version rather than set a skip and forget to ever run the test.

Reply:
Maybe we should run the test only on CI? (by a environment variable or an another way that unittest support)

On the one hand, we'll see when our tests break. On the other hand, a user will not be confused with a failed test after updating Tarantool.

Reply:
> On the other hand, a user will not be confused with a failed test after updating Tarantool.

I think that when one should get this fail, it would be the moment we need to replace xfail with conditional xfail based on version.

Reply:
Add a task to a sprint, fix it, and then release a new version... It may take a long time for users to get a working version.

It's not a good idea to make CI easier for us in the moment, but make life harder for users in the future.

Reply:
I don't get how choosing between `xfail` and `skip` affects user with release versions.

Reply:
If you are running module tests, you are a developer. As a developer, you may either ignore a couple of failing tests after you find out what's the reason (it shouldn't be complicated since relevant issue link is provided), or report the issue to us (to be honest, I'm sure that developer would already be one of us), or fix it (it's an hour long work together with all required stuff like commit messages and PR).

Reply:
I would be glad to provide `skip` tests with conditions, but these conditions not yet exist. Make a test that always will skip until one remembers to "uncomment it" is also doesn't seems like a good approach to me.